packages/core/src/collision/CollisionDetection.ts

```diff
@@ -55,43 +47,54 @@ export class CollisionDetection extends Script {
         for (let i = 0, len = colliders.length; i < len; i++) {
           const collider = colliders[i];
           if (collider != this._myCollider && this._boxCollision(collider)) {
-            overlopCollider = collider;
-            // @ts-ignore
-            this.dispatch("collision", { collider });
+            overlappedCollider = collider;
+            let scripts: Script[] = [];
+            this.entity.getComponents(Script, scripts);
+            scripts.forEach((script) => {
+              script.onTriggerStay(collider);
+            });
           }
         } // end of for
       } else if (this._myCollider instanceof ASphereCollider) {
         this._sphere = this._getWorldSphere(this._myCollider);
         for (let i = 0, len = colliders.length; i < len; i++) {
           const collider = colliders[i];
           if (collider != this._myCollider && this._sphereCollision(collider)) {
-            overlopCollider = collider;
-            // @ts-ignore
-            this.dispatch("collision", { collider });
+            overlappedCollider = collider;
+            let scripts: Script[] = [];
+            this.entity.getComponents(Script, scripts);
+            scripts.forEach((script) => {
+              script.onTriggerStay(collider);
+            });
           }
         } // end of for
       }
     } // end of if
 
-    //-- overlop events
-    if (overlopCollider != null && this._overlopCollider != overlopCollider) {
-      // @ts-ignore
-      this.dispatch("begin_overlop", { collider: overlopCollider });
+    //-- overlap events
+    if (overlappedCollider != null && this._overlappedCollider != overlappedCollider) {
+      let scripts: Script[] = [];
+      this.entity.getComponents(Script, scripts);
+      scripts.forEach((script) => {
+        script.onTriggerEnter(overlappedCollider);
+      });
     }
 
-    if (this._overlopCollider != null && this._overlopCollider != overlopCollider) {
-      const e = this._overlopCollider;
-      // @ts-ignore
-      this.dispatch("end_overlop", { collider: e });
+    if (this._overlappedCollider != null && this._overlappedCollider != overlappedCollider) {
+      let scripts: Script[] = [];
+      this.entity.getComponents(Script, scripts);
```

Comment:
call `getComponets() every frame is very slow`, you can use internal field `this.entity._scripts` directly.